### PR TITLE
Add IPFS Update instructions to the Install section.

### LINKED ISSUE #300: Create a guide/section on using IPFS Update.
There is currently no documentation for using `ipfs-update` within the `ipfs/docs` repo. There are instructions on the [`ipfs/ipfs-update` repo](https://github.com/ipfs/ipfs-update/blob/v1.6.0/README.md), however. These instructions should be brought into the `ipfs/docs` repo and updated. These docs can go into `docs.ipfs.io/how-tos/ipfs-update/`. The section should cover:

1. Install `ipfs-update`.
1. Use `ipfs-update` to install IPFS.
1. Use `ipfs-update` to update your current IPFS installation.
1. Use `ipfs-update` to downgrade your IPFS installation.
1. Uninstall `ipfs-update`.

### PR DESCRIPTION
**Pull request by alexmmueller**
It is not completely ready by now, but I thought I get your comments as its almost finished.

* Uninstalll commands for windows are missing, but section is prepared. Maybe you can add them?
* I tried to use the big OS icon links in the `ipfs-update` section, but due to OS same section names it is not possible to link to the correct sections. I skipped it for now. Please let me know if you prefer something els
* I used `curl` in instructions for macOS as `wget` is not preinstalled on macOs (at least not on my boxes), but `curl` is
* I added `sudo` to the commands where root privileges are required

Feedback is welcome :)

Admin edit: Closes #300

### CODE REVIEW

File: docs/install/command-line.md [hunk 1]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+# You need to CD to a directory _outside_ of your GOPATH.
+$ cd /
+# Install with go modules enabled
+$ GO111MODULE=on go get github.com/ipfs/ipfs-update
+```
+
+Note: Your $GOPATH/bin should be within $PATH for the result ipfs-update binary
+to be found.
+
+### Use `ipfs-update` to install & update Go-IPFS
```
Comment (johnnymatthews) [inline docs/install/command-line.md:302]:
```suggestion
### Install using IPFS Update
```

File: docs/install/command-line.md [hunk 2]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
+```bash
+$ ipfs-update install <version>
+```
```
Comment (johnnymatthews) [inline docs/install/command-line.md:306]:
````suggestion
Run `ipfs-update install` followed by the version of Go-IPFS you want to install:

```bash
ipfs-update install 0.5.0
```

To install the latest release of Go-IPFS use the `latest` tag:

```bash
ipfs-update install latest
```

````

File: docs/install/command-line.md [hunk 3]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
+Downloads, tests, and installs the specified version (or "latest" for latest version) of Go-IPFS. The existing version is stashed in case a revert is needed.
```
Comment (johnnymatthews) [inline docs/install/command-line.md:308]:
```suggestion
`ipfs-update install` downloads, tests, and installs the specified version of Go-IPFS. If a version of IPFS is already installed, that version is _stashed_ and can be reverted to later.
```

File: docs/install/command-line.md [hunk 4]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
+Note: In order to update Go-IPFS just install the new version. Check See the [`ipfs/ipfs-update` GitHub repository for details →](https://github.com/ipfs/ipfs-update)
```
Comment (johnnymatthews) [inline docs/install/command-line.md:310]:
```suggestion
```

File: docs/install/command-line.md [hunk 5]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
+### Use `ipfs-update` to downgrade Go-IPFS
```
Comment (johnnymatthews) [inline docs/install/command-line.md:312]:
```suggestion
### Downgrade using IPFS Update
```

File: docs/install/command-line.md [hunk 6]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
```
Comment (johnnymatthews) [inline docs/install/command-line.md:313]:
```suggestion
Use the `revert` function to roll-back to a previous version of Go-IPFS.

```

File: docs/install/command-line.md [hunk 7]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+```bash
+$ ipfs-update revert
```
Comment (johnnymatthews) [inline docs/install/command-line.md:315]:
```suggestion
ipfs-update revert
```

File: docs/install/command-line.md [hunk 8]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+```
+
+Reverts to the previously installed version of Go-IPFS. This is useful if the newly installed version has issues and you would like to switch back to your older stable installation.
```
Comment (johnnymatthews) [inline docs/install/command-line.md:318]:
```suggestion
`ipfs-update revert` reverts to the previously installed version of Go-IPFS. This is useful if the newly installed version has issues and you would like to switch back to your older stable installation.
```

File: docs/install/command-line.md [hunk 9]
```diff
@@ -155,9 +155,211 @@ The IPFS team manages the [dist.ipfs.io website](https://dist.ipfs.io/) to help
...
+
+### Uninstall `ipfs-update`
+
+Uninstalling `ipfs-update` involves removing the binary and adjusting the `PATH` variable on Windows systems.
```
Comment (johnnymatthews) [inline docs/install/command-line.md:322]:
```suggestion
To uninstall IPFS Update, delete the binary and `ipfs-update` from your `PATH` variable.
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
This is a great start! However, I think it might be better if we moved everything in here into it's own page. Somewhere like `install/ipfs-update`. This `install/command-line` page is already getting pretty long.

Carry on with your work here though, and I'll move things around once you've finished :)

Also, I've converted this PR to a draft since you're still working on it.

**Comment by alexmmueller** [comment]
I accepted all changes you proposed @johnnymatthews. I am a little unsre of the value of having powershell command for updating `PATH`as this is not trival it seams and requires a small program (get variable, iterate over it, etc). Finding a file is a little easier: Can you check if the powershell command:

gci -recurse -filter "ipfs-update.exe" -File -ErrorAction SilentlyContinue

Will work? If not, I will setup a VM to try this out.

**Comment by johnnymatthews** [comment]
Sounds good. If you've got time to explore the Powershell section through a VM then that'd be great! No worries if you can't though. I've got a Windows machine, but it's out of action right now. So it'll be a couple of days until I can get into it.

> Could I have create the PR as a draft as well?

Yeah! I didn't realize this for a while, but there's a drop down menu when you click **Create pull request** that lets you create the PR as a draft. [It's kinda hidden](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

**Comment by johnnymatthews** [comment]
Looks like my Windows machine is gonna be out of action for a while, so I'll boot up a VM and get the Powershell stuff tested.

**Comment by alexmmueller** [comment]
> Were you able to boot up a Windows VM @alexmmueller?

After some painful hours, yes :) I guess I will be able to get everything tested within the next weekend. At the moment I am not sure whether to list the commands for removing a variable from PATH in powershell or go for the instructions on how to remove the variable via the settings widget in windows. Removing the directory from PATH via powershell is not a one liner. Do you have any preference @johnnymatthews ?

**Comment by johnnymatthews** [comment]
@alexmmueller you can ignore instructions for removing the PATH variable for Windows. There's a bunch of ways to do it, and they all have several steps involved. Instead we should link to the [Microsoft documentation on how to update/edit/delete your PATH](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160).

**Comment by alexmmueller** [comment]
I reviewed the windows instructions today and found some issues:
* the PowerShell profile file does not necessarily exist --> I added a check and profile creation instructions
* PowerShell does by default block profile files --> I added a note and a link to the respective documentation on PowerShell ExecutionPolicies

If you want anything changed, let me know.

If you like the changes and there are no change requests, I think we should also adopt the windows PATH part for GO-IPFS accordingly. Do you agree?

File: docs/install/command-line.md [hunk 10]
```diff
@@ -321,20 +339,18 @@ To uninstall IPFS Update, delete the binary and `ipfs-update` from your `PATH` v
...
-
+   Remove-Item -Recurse -Force C:\Users\Johnny\Apps\ipfs-update_v1.6.0
    ```
 
-3. Update the `PATH` variable
-
-   ```powershell
-
-   ```
+3. Update the `PATH` variable as described in the [Microsoft documentation on how to update/edit/delete your PATH](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160)
```
Comment (johnnymatthews) [inline docs/install/command-line.md]:
```suggestion
3. Update your systems `PATH` variable. This process differs between Windows installations, so please check the [Microsoft documentation for details](https://docs.microsoft.com/en-us/cpp/build/setting-the-path-and-environment-variables-for-command-line-builds?view=msvc-160).
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
Everything looks really good here! Just one minor change.

**Comment by alexmmueller** [comment]
I did the requested change. Let me know if anything else has to be done here.

**Review by johnnymatthews — APPROVED** [review]

**Comment by johnnymatthews** [comment]
Everything looks good here @alexmmueller, thanks for your efforts! I'm gonna move things around a little bit, and then get things merged in.

**Pull request merged**